src/server/JasmineGraphServer.cpp

```diff
@@ -3042,44 +3058,37 @@ void JasmineGraphServer::initiateFiles(std::string graphID, std::string training
 void JasmineGraphServer::initiateCommunication(std::string graphID, std::string trainingArgs, SQLiteDBInterface sqlite) {
 
-    Utils utils;
-    int fl_clients = stoi(utils.getJasmineGraphProperty("org.jasminegraph.fl_clients"));
-    int threadLimit = fl_clients + 1;
-    std::thread *workerThreads = new std::thread[threadLimit];
+    int count = 0;
+    JasmineGraphTrainingSchedular *schedular = new JasmineGraphTrainingSchedular();
+    map<string, map<int, int>> scheduleForAllWorkers = schedular->schedulePartitionTraining(graphID);
+    std::map<std::string, JasmineGraphServer::workerPartition> graphPartitionedHosts = this->getWorkerPartitions(
+            graphID);
+    int partition_count = graphPartitionedHosts.size();
 
-    Utils::worker workerInstance;
-    vector<Utils::worker> workerVector = utils.getWorkerList(sqlite);
+    std::cout << partition_count << std::endl;
 
+    Utils utils;
     trainingArgs = trainingArgs;
-    int threadID = 0;
-
-    for (int i = 0; i < workerVector.size(); i++) {
-
-        workerInstance = workerVector[i];
-
-        int serverPort = stoi(workerInstance.port);
-        int serverDataPort = stoi(workerInstance.dataPort);
-
-        if (i==0) {
 
-            workerThreads[threadID] = std::thread(initiateServer,"localhost", serverPort,
-                                                    serverDataPort,trainingArgs,fl_clients, to_string(i));
-            threadID++;
-        }
-
-        workerThreads[threadID] = std::thread(initiateClient,"localhost", serverPort, serverDataPort,trainingArgs +
-                                                    " " + to_string(i), fl_clients, to_string(i));
-        threadID++;
+    int fl_clients = stoi(utils.getJasmineGraphProperty("org.jasminegraph.fl_clients"));
 
-    }
+    count = 1;
+    std::map<std::string, JasmineGraphServer::workerPartition>::iterator j = graphPartitionedHosts.begin();
+    JasmineGraphServer::workerPartition workerPartition = j->second;
 
-    workerThreads[0].join();
+    initiateServer(workerPartition.hostname, workerPartition.port,
+                   workerPartition.dataPort, trainingArgs,fl_clients);
+    for (; j != graphPartitionedHosts.end(); j++) {
+        workerPartition = j->second;
+        map<int, int> scheduleOfHost = scheduleForAllWorkers[workerPartition.hostname];
+        int iterationOfPart = scheduleOfHost[stoi(workerPartition.partitionID)];   // todo check why schedule was not being used in the original implementation
 
-    for (int threadCount = 1; threadCount < threadLimit; threadCount++) {
-        workerThreads[threadCount].join();
+        initiateClient(workerPartition.hostname, workerPartition.port,
+                       workerPartition.dataPort, trainingArgs + " " + workerPartition.partitionID, iterationOfPart, std::to_string(partition_count));
+        count++;
     }
 
-    server_logger.log("Federated learning commands sent", "info");
+    std::cout << "Federated learning commands sent" <<std::endl;
```

Comment:
Should remove `std::out` statements. Only log if required into log file.

Reply:
fixed

Reply:
Ok